File: c++/e798.cpp
#include <iostream>
using namespace std;

int a[21][21];

int fourmax(int x, int y){
    int temp=a[x][y];
    for (int i = 0; i <2;i++){
        for(int j = 0; j <2;j++){
            if (temp<a[i+x][j+y]) temp=a[i+x][j+y];
        }
    }
    return temp;
}

int main()
{
    int t ;
    while (cin >> t){
        for (int i=0;i<t;i++)
            for (int j=0;j<t;j++) cin >> a[i][j];
        
        for (int i=0;i<t;i+=2){
            for (int j=0;j<t;j+=2){
                cout << fourmax(i,j) << ' ';
            }
            cout << endl;
        }
    }
}

File: python/e798.py
a=[]

def fourmax(x,y):
    temp=a[x][y]
    for i in range(2):
        for j in range(2):
            if (a[i+x][j+y]>temp): temp = a[i+x][j+y]
    return temp

n =int(input())
for i in range(n): a.append(list(map(int,input().replace('\n','').split())))
for i in range(0,n,2):
    for j in range(0,n,2):
        print(fourmax(i,j),end=' ')
    print()